File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/HitWallEvent.java
package dev.robocode.tankroyale.botapi.events;

/**
 * Event occurring when your bot has hit a wall.
 */
@SuppressWarnings("unused")
public final class HitWallEvent extends BotEvent {

    /**
     * Initializes a new instance of the BotHitWallEvent class.
     *
     * @param turnNumber is the turn number when the bot has hit the wall.
     */
    public HitWallEvent(int turnNumber) {
        super(turnNumber);
    }
}

File: bot-api/dotnet/api/src/events/HitWallEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events;

/// <summary>
/// Event occurring when the bot has hit a wall.
/// </summary>
public sealed class HitWallEvent : BotEvent
{
    /// <summary>
    /// Initializes a new instance of the BotHitWallEvent class.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    [JsonConstructor]
    public HitWallEvent(int turnNumber) : base(turnNumber)
    {
    }
}